### PR TITLE
HTTP Client errors docs for iOS

### PR DESCRIPTION
**Pull request by marandaneto**
Relates to https://github.com/getsentry/sentry-cocoa/pull/2308
Depends on https://github.com/getsentry/publish/issues/1534

### CODE REVIEW

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+title: HTTP Client Errors
+sidebar_order: 13
+description: "This feature, once enabled, automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry"
+---
+
+This feature, once enabled, automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry. The error event will contain the `request` and `response` data, such as `url`, `status_code`, and so on.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:7]: Rearranging this to cut down on commas

```suggestion
Once enabled, this feature automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry. The error event will contain the `request` and `response` data, such as `url`, `status_code`, and so on.
```

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+This feature is opt-in and can be enabled by setting the `enableCaptureFailedRequests` option to `true`:
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:9]: Avoiding the repetition of "this feature" as the beginning of three sections in a row.

```suggestion
Sending HTTP client errors is an opt-in feature and can be enabled by setting the `enableCaptureFailedRequests` option to `true`:
```

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    SentryHttpStatusCodeRange *httpStatusCodeRange =
+            [[SentryHttpStatusCodeRange alloc] initWithMin:400 max:599];
+    options.failedRequestStatusCodes = @[ httpStatusCodeRange ];
+}];
+```
+
+HTTP client errors from every target (`.*` regular expression) are automatically captured, but you can change this behavior by setting the `failedRequestTargets` option with either a regular expression or a plain `String`. A plain string must contain at least one of the items from the list. Plain strings don't have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:54]:
```suggestion
HTTP client errors from every target (`.*` regular expression) are automatically captured, but you can change this behavior by setting the `failedRequestTargets` option with either a regular expression or a plain `String`. A plain string must contain at least one of the items from the list. Plain strings don't have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option:
```

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+```objc {tabTitle:Objective-C}
+@import Sentry;
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    options.failedRequestTargets = @[ @"www.example.com" ];
+}];
+```
+
+Error events may contain PII data, such as `Headers` and `Cookies`, Sentry already does data scrubbing by default, but you can scrub any data before it is sent, learn more about it in the [Scrubbing Sensitive Data docs](/platforms/apple/guides/ios/data-management/sensitive-data/).
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:74]:
```suggestion
Error events may contain PII data, such as `Headers` and `Cookies`. Sentry already does data scrubbing by default, but you can scrub any data before it is sent. Learn more in [Scrubbing Sensitive Data](/platforms/apple/guides/ios/data-management/sensitive-data/).
```

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+Those events are searchable and you can set alerts on them if you use the `http.url` and `http.status_code` properties. Learn more in our full [Searchable Properties](/product/sentry-basics/search/searchable-properties/) documentation.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:76]:
```suggestion
These events are searchable and you can set alerts on them if you use the `http.url` and `http.status_code` properties. Learn more in our full [Searchable Properties](/product/sentry-basics/search/searchable-properties/) documentation.
```

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+```objc {tabTitle:Objective-C}
+@import Sentry;
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    options.enableCaptureFailedRequests = YES;
+}];
+```
+
+This feature requires the setting `enableSwizzling` to be enabled as well.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:29]:
```suggestion
To use this feature, you must also enable the `enableSwizzling` setting.
```

File: src/platforms/common/configuration/options.mdx [hunk 7]
```diff
@@ -507,6 +507,18 @@ _(New in version 6.6.0)_
 
 </ConfigKey>
 
+<ConfigKey name="capture-failed-requests" supported={["apple"]}>
+
+This feature, once enabled, automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry.
```
Comment (imatwawana) [inline src/platforms/common/configuration/options.mdx:512]:
```suggestion
Once enabled, this feature automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry.
```

**Review by imatwawana — COMMENTED** [review]
Made some wording tweaks

**Review by imatwawana — APPROVED** [review]

**Comment by marandaneto** [comment]
@philipphofmann @brustolin review here would be nice, thanks.

File: src/platforms/apple/common/configuration/http-client-errors.mdx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+```objc {tabTitle:Objective-C}
+@import Sentry;
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    options.enableCaptureFailedRequests = YES;
+}];
+```
+
+To use this feature, you must also enable the `enableSwizzling` setting.
```
Comment (philipphofmann) [inline src/platforms/apple/common/configuration/http-client-errors.mdx:29]: `m`: `enableSwizzling` is enabled by default. I think we don't need to mention it here.

File: src/platforms/apple/common/configuration/swizzling.mdx [hunk 9]
```diff
@@ -12,6 +12,7 @@ __macOS__
 - <PlatformLink to="/performance/instrumentation/automatic-instrumentation/#file-io-tracking">Auto instrumentation for File I/O operations</PlatformLink>
 - <PlatformLink to="/performance/instrumentation/automatic-instrumentation/#core-data-tracking">Auto instrumentation for Core Data operations</PlatformLink>
 - <PlatformLink to="/performance/connect-services/">Automatically added sentry-trace header to HTTP requests for distributed tracing</PlatformLink>
+- <PlatformLink to="/configuration/http-client-errors/">HTTP Client Errors</PlatformLink>
```
Comment (philipphofmann) [inline src/platforms/apple/common/configuration/swizzling.mdx:15]: 💯 for adding it here.

**Review by philipphofmann — COMMENTED** [review]
Please also add the feature to the following locations:
https://github.com/getsentry/sentry-docs/blob/431dee65cc99bd4f4c0e3d756e2de2c68798a52c/src/platform-includes/getting-started-config/apple.mdx?plain=1#L15-L18

https://github.com/getsentry/sentry-docs/blob/431dee65cc99bd4f4c0e3d756e2de2c68798a52c/src/platform-includes/getting-started-config/apple.mdx?plain=1#L34-L37

https://github.com/getsentry/sentry-docs/blob/431dee65cc99bd4f4c0e3d756e2de2c68798a52c/src/platform-includes/getting-started-config/apple.mdx?plain=1#L56-L59

And also the wizard please

https://github.com/getsentry/sentry-docs/blob/431dee65cc99bd4f4c0e3d756e2de2c68798a52c/src/wizard/apple/ios.md?plain=1#L41-L44

https://github.com/getsentry/sentry-docs/blob/431dee65cc99bd4f4c0e3d756e2de2c68798a52c/src/wizard/apple/ios.md?plain=1#L67-L70

**Comment by philipphofmann** [comment]
🥇 Thank you @marandaneto for also creating the docs for this feature.

**Pull request merged**